.infrastructure/hpa.yml

```diff
@@ -0,0 +1,24 @@
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: todoapp
+  namespace: mateapp
+spec:
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: todoapp
+  minReplicas: 2
+  maxReplicas: 5
+  metrics:
+  - type: Resource
+    resource:
+      name: cpu
+      target:
+        type: Utilization
+        averageUtilization: 70
+  - type: Resource
+    resource:
+      name: memory
+      target: Utilization
+      averageUtilization: 70
```

Comment:
The 'target' field for the memory resource should include `type: Utilization` to match the structure used for the CPU resource. This ensures that the HPA functions as expected.